### DISCUSSION TITLE
Curious about targets

### DISCUSSION THREAD

**Discussion by goodcodedev** [BosqueLanguage/BosqueCore#156] [General]
Curious about the thinking on targets for Bosque.
I know there are interesting things to explore wrt. own runtime.
But will javascript continue to be a prominent target, and is it expected to be a fairly lean and well optimized target.
And is Wasm on the radar?

**Comment by marron-at-work** [comment]
Currently the main target is JavaScript, mainly as a easy to run everywhere and debug option. But I do expect it to continue as a prominent target, even as work on the [native runtime](https://bosquelanguage.github.io/2024/10/01/an-omega-c-runtime.html) starts, and the emitter is setup to output readable code when possible. 

I am not sure about trying for WASM support. I don't really see the value add beyond a nice JS backend. The item I was thinking about was providing support for emitting a full NPM package (package.json and all) from a Bosque project to make Node.js interop easy.
